File: rails-api/spec/requests/olympians_index_spec.rb
require 'rails_helper'

describe 'GET api/v1/olympians' do
  it 'returns an index of all olympians' do
    oly1, oly2 = create(:olympian), create(:olympian)

    get '/api/v1/olympians'

    expect(response.status).to eq(200)

    data = JSON.parse(response.body, symbolize_names: true)

    expected = { 
      olympians: [
        {
          id: oly1.id,
          name: oly1.name,
          sex: oly1.sex,
          age: oly1.age,
          height: oly1.height,
          weight: oly1.weight
        },
        {
          id: oly2.id,
          name: oly2.name,
          sex: oly2.sex,
          age: oly2.age,
          height: oly2.height,
          weight: oly2.weight
        }
      ]
    }

    expect(data).to eq(expected)
  end
end

File: express/tests/olympiansIndex.spec.js
const env = process.env.NODE_ENV || 'development';
const config = require('../knexfile')[env];
const DB = require('knex')(config);
const request = require("supertest");
const app = require('../app')


describe('api/v1/olympians', () => {
  beforeEach(async () => {
    await DB.raw('TRUNCATE TABLE olympians')
  })

  afterEach(async () => {
    await DB.raw('TRUNCATE TABLE olympians')
  })

  it('returns an index of all olympians', async () => {
    oly1 = await DB('olympians')
      .insert({ name: 'Bob', sex: 'M', age: 25, height: 170, weight: 65 })
      .returning('*')

    oly2 = await DB('olympians')
      .insert({ name: 'Sue', sex: 'F', age: 22, height: 160, weight: 54 })
      .returning('*')
    
    const expected = {
      olympians: [ oly1, oly2 ]
    }

    const response = await request(app).get('/api/v1/olympians')
    console.log(response.body)
    expect(response.status).toBe(200);
    expect(response.body).toEqual(expected);
  })
})